modules/jwplayerBidAdapter.js

```diff
@@ -50,4 +90,111 @@ export const spec = {
   // onBidderError: function({ error, bidderRequest }) {}
 };
 
+function buildRequest(bidRequest, bidderRequest) {
+  const openrtbRequest = {
+    id: bidRequest.bidId,
+    imp: buildRequestImpression(bidRequest, bidderRequest),
+    site: buildRequestSite(bidRequest),
+    device: buildRequestDevice()
+  };
+
+  // Attaching GDPR Consent Params
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(openrtbRequest, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(openrtbRequest, 'regs.ext.gdpr', (bidderRequest.gdprConsent.gdprApplies ? 1 : 0));
+  }
+
+  // CCPA
+  if (bidderRequest.uspConsent) {
+    deepSetValue(openrtbRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+
+  return JSON.stringify(openrtbRequest);
+}
+
+function buildRequestImpression(bidRequest) {
+  const impressionObject = {
+    id: bidRequest.adUnitCode,
+  };
+
+  impressionObject.video = buildImpressionVideo(bidRequest);
+
+  const bidFloorData = buildBidFloorData(bidRequest);
+  if (bidFloorData) {
+    impressionObject.bidfloor = bidFloorData.floor;
+    impressionObject.bidfloorcur = bidFloorData.currency;
+  }
+
+  impressionObject.ext = buildImpressionExtension(bidRequest);
+
+  return [impressionObject];
+}
+
+function buildImpressionVideo(bidRequest) {
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  const video = {};
+
+  VIDEO_ORTB_PARAMS.forEach((param) => {
+    if (videoParams.hasOwnProperty(param)) {
+      video[param] = videoParams[param];
+    }
+  });
+
+  return video;
+}
+
+function buildImpressionExtension(bidRequest) {
+  return {
+    appnexus: {
+      placement_id: bidRequest.params.placementId
+    }
+  };
+}
+
+function buildBidFloorData(bidRequest) {
+  const {params} = bidRequest;
+  const currency = params.currency || 'USD';
+
+  let floorData;
+  if (isFn(bidRequest.getFloor)) {
+    const bidFloorRequest = {
+      currency: currency,
+      mediaType: VIDEO,
+      size: '*'
+    };
+    floorData = bidRequest.getFloor(bidFloorRequest);
+  } else if (params.bidfloor) {
+    floorData = {floor: params.bidfloor, currency: currency};
+  }
+
+  return floorData;
+}
+
+function buildRequestSite(bidRequest) {
+  const configSite = config.getConfig('ortb2.site') || {};
+
+  const site = {
+    domain: config.publisherDomain || configSite.domain || window.location.hostname,
+    page: config.pageUrl || configSite.page || window.location.href,
+  };
+
+  const referer = bidRequest.refererInfo && bidRequest.refererInfo.referer;
+  if (referer) {
+    site.ref = referer;
+  }
+
+  // Site Content
+  if (configSite && configSite.content && isPlainObject(configSite.content)) {
+    site.content = configSite.content;
+  }
+  return site;
```

Comment:
I think we went the wrong direction with this. The publisher know's best about what to put in the `site` object since they own the page. So we should use what they set, and fallback where we think we know best. So, I propose:
```
function buildRequestSite(bidRequest) {
  const site = config.getConfig('ortb2.site') || {};
  site.domain = site.domain || config.publisherDomain || window.location.hostname;
  site.page = site.page || config.pageUrl || window.location.href;
  const referer = bidRequest.refererInfo && bidRequest.refererInfo.referer;
  if (!site.ref && referer) {
    site.ref = referer;
  }

  return site;
```

Reply:
done!